grammar.js

```diff
@@ -860,21 +903,139 @@ module.exports = grammar({
       ),
     ),
 
+    hive_location: $ => prec.right(
+        seq(
+            $.keyword_location,
+            field('path', alias($._literal_string, $.identifier)),
+            optional(
+                seq(
+                    $.keyword_cached,
+                    $.keyword_in,
+                    field('pool', alias($._literal_string, $.identifier)),
+                    optional(
+                        choice(
+                            $.keyword_uncached,
+                            seq(
+                                $.keyword_with,
+                                seq(
+                                    $.keyword_replication,
+                                    '=',
+                                    field('value', alias($._number, $.literal)),
+                                ),
+                            ),
+                        ),
+                    ),
+                )
+            )
+        ),
+    ),
+
+    row_format: $ => seq(
+        $.keyword_row,
+        $.keyword_format,
+        $.keyword_delimited,
+        optional(
+            seq(
+                $.keyword_fields,
+                $.keyword_terminated,
+                $.keyword_by,
+                $._literal_string,
+                optional(
+                    seq(
+                        $.keyword_escaped,
+                        $.keyword_by,
+                        $._literal_string,
+                    )
+                )
+            )
+        ),
+        optional(
+            seq(
+                $.keyword_lines,
+                $.keyword_terminated,
+                $.keyword_by,
+                $._literal_string,
+            )
+        )
+    ),
+
+    table_sort: $ => seq(
+        $.keyword_sort,
+        $.keyword_by,
+        seq(
+            '(',
+                $.identifier,
+                repeat(seq(',', ($.identifier))),
+            ')',
+        )
+    ),
+
+    table_partition: $ => seq(
+        choice(
+            // Postgres/MySQL style
+            seq(
+                $.keyword_partition,
+                $.keyword_by,
+                choice(
+                    $.keyword_range,
+                    $.keyword_hash,
+                )
+            ),
+            // Hive style
+            seq(
+                $.keyword_partitioned,
+                $.keyword_by,
+            ),
+        ),
+        choice(
+            seq('(', $.identifier, ')'), // postgres
+            $.column_definitions // impala/hive
+        )
+    ),
+
+    hive_stored_as: $ => seq(
```

Comment:
I wonder if we should avoid named nodes with specific dialect names.

e.g. rename this to `stored_as` unless we have a conflict.

Reply:
yeah, I was unsure about this as well. I will remove the dialect hint from the name.

Reply:
Alternatively, I could make these nodes anonymous like `_mysql_update_statement` and `_postgres_update_statement`